ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/ProfileStatsWidget.tsx

```diff
@@ -0,0 +1,129 @@
+    isLoading: boolean;
+    error: Error | undefined;
+    profileScanStats: ComplianceProfileScanStats | undefined;
+};
+
+function ProfileStatsWidget({ error, isLoading, profileScanStats }: ProfileStatsWidgetProps) {
+    const [widgetContainer, setWidgetContainer] = useState<HTMLDivElement | null>(null);
+    const widgetContainerResizeEntry = useResizeObserver(widgetContainer);
+
+    if (isLoading) {
+        return (
+            <Bullseye>
+                <Spinner />
+            </Bullseye>
+        );
+    }
+
+    if (error) {
+        return (
+            <Bullseye>
+                <EmptyStateTemplate
+                    title="Error loading profile stats"
+                    headingLevel="h3"
+                    icon={ExclamationCircleIcon}
+                    iconClassName="pf-v5-u-danger-color-100"
+                >
+                    {getAxiosErrorMessage(error.message)}
+                </EmptyStateTemplate>
+            </Bullseye>
+        );
+    }
+
+    if (profileScanStats) {
+        const { passCount, failCount, manualCount, otherCount, totalCount } = getStatusCounts(
+            profileScanStats.checkStats
+        );
+
+        const data: ChartData[] = [
+            {
+                x: 'Passing',
+                y: passCount / totalCount,
+                color: PASSING_COLOR,
+            },
+            {
+                x: 'Failing',
+                y: failCount / totalCount,
+                color: FAILING_COLOR,
+            },
+            {
+                x: 'Manual',
+                y: manualCount / totalCount,
+                color: MANUAL_COLOR,
+            },
+            {
+                x: 'Other',
+                y: otherCount / totalCount,
+                color: OTHER_COLOR,
+            },
+        ];
+
+        return (
+            <div ref={setWidgetContainer}>
+                <Chart
+                    ariaDesc="Aging images grouped by date of last update"
+                    ariaTitle="Aging images"
+                    domainPadding={{ x: [50, 50] }}
+                    padding={{ top: 30, bottom: 65, left: 65, right: 20 }}
+                    height={defaultChartHeight}
+                    width={widgetContainerResizeEntry?.contentRect.width ?? 0}
+                    containerComponent={<ChartContainer responsive />}
+                >
+                    <ChartAxis label="Check status" />
+                    <ChartAxis
+                        tickFormat={(t) => `${Math.round(t * 100)}%`}
+                        domain={[0, 1]}
+                        dependentAxis
+                        showGrid
+                    />
+                    <ChartBar
+                        key="testing"
+                        barWidth={defaultChartBarWidth}
+                        data={data}
+                        style={{
+                            data: {
+                                fill: ({ datum }: DatumArgs) => (datum ? datum.color : 'gray'),
```

Comment:
Is `: 'gray'` defensive programming? I backed up when I read `datum.y` below without conditional.

What do you think about one of the constant variables instead of `'gray'` literal?

If needed (and when in doubt, keep) which color convention does it correspond to?
* icon colors as above?
* label colors?

Reply:
It's set that way to satisfy typescript, I'm not sure how to work around the issue without have access to `Victory` types which is what patternfly is using under the hood. I think using the `OTHER_COLOR` makes sense as a "backup" though.